chore: docs for marmalade client utils (#2234)
kadena-community/kadena.js: packages/libs/client-utils/src/marmalade/README.md

---

> title: kadena marmalade client-utils

Reviewer: ```suggestion
title: Marmalade client-utils
```

description:
  The @kadena/client-utils library provides a TypeScript-based API for interacting
  with marmalade smart contracts on the Kadena network.
menu: Reference

> label: Kadena Marmalade Client Utils

Reviewer: ```suggestion
label: Marmalade client utilities
```

order: 5
layout: full
tags: ['TypeScript', 'Kadena', 'Kadena client utils', 'frontend', 'backend', 'marmalade', 'client-utils', 'utils']
---

> # Kadena Marmalade Client Utils

Reviewer: ```suggestion
# Marmalade client utilities
```

    },
    chainId: "0",
  },
  config
);
```

### Update URI

The `updateUri` function enables the updating of a token's URI, modifying its associated metadata.

```typescript
updateUri(inputs, config): IEmitterWrapper
```

#### Inputs

| Parameter    | Type                      | Description                 |
| ------------ | ------------------------- | --------------------------- |
| policyConfig | object                    | concrete policies used      |
| tokenId      | string                    | target token                |
| uri          | string                    | sender account with guard   |
| guard        | object                    | uriGuard account and guard  |
| chainId      | ChainId                   | origin                      |

#### Output
```typescript
boolean
```

#### Example

```typescript
const result = await updateUri({
    tokenId: "t:...",
    uri: "ipfs://new-uri",
    guard: {
      account: "k:368820f80c324bbc7c2b0610688a7da43e39f91d118732671cd9c7500ff43cca",
      keyset: {
        keys: ["368820f80c324bbc7c2b0610688a7da43e39f91d118732671cd9c7500ff43cca"],
        pred: 'keys-all'
      }
    },
    chainId: "0",
  },
  config
);
```

> 
> ### Withdraw Token
> 
> The `withdrawToken` function allows a seller to withdraw a token from a sale.

Reviewer: ```suggestion
## withdrawToken

Use `withdrawToken` to withdraw a specified token from a sale.
```

```typescript
withdrawToken(inputs, config): IEmitterWrapper
```

> #### Inputs

Reviewer: ```suggestion
### Inputs
```

| Parameter    | Type         | Description                 |
| ------------ | ------------ | --------------------------- |
| policyConfig | object       | concrete policies used      |
| tokenId      | string       | target token                |
| saleId       | string       | target sale                 |
| amount       | IPactDecimal | amount to withdraw          |
| timeout      | IPactInt     | sale timeout                |
| seller       | object       | seller account with guard   |
| chainId      | ChainId      | origin                      |

> #### Output

Reviewer: ```suggestion
### Output

```

```typescript
boolean
```

> #### Example

Reviewer: ```suggestion
### Example
```
